numpy/lib/stride_tricks.py

```diff
@@ -35,7 +35,19 @@ def _maybe_view_as_subclass(original_array, new_array):
     return new_array
 
 
-def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
+def _actual_range(x):
+    if not isinstance(x, np.ndarray):
```

Comment:
`x` is guaranteed to be an `ndarray` instance here, so this test can be omitted.